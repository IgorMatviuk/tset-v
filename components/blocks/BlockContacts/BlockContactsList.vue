<template>
  <ul class="contacts-list">
    <li class="contacts-list__item">
      <svg-icon
        name="phone"
        class="contacts-list__icon"
        width="16"
        height="16"
      />
      <span class="contacts-list__description">
        <span class="contacts-list__title">
          {{ $t('contacts.titles.phone') }}
        </span>
        <a
          :href="`tel:${contacts.phone}` | convertPhone"
          class="contacts-list__link"
        >
          {{ contacts.phone }}
        </a>
      </span>
    </li>
    <li class="contacts-list__item">
      <svg-icon
        name="mail"
        class="contacts-list__icon"
        width="16"
        height="16"
      />
      <span class="contacts-list__description">
        <span class="contacts-list__title">
          {{ $t('contacts.titles.email') }}
        </span>
        <a :href="`mailto:${contacts.email}`" class="contacts-list__link">
          {{ contacts.email }}
        </a>
      </span>
    </li>
    <li class="contacts-list__item">
      <svg-icon
        name="headset"
        class="contacts-list__icon"
        width="16"
        height="16"
      />
      <span class="contacts-list__description">
        <span class="contacts-list__title">
          {{ $t('contacts.titles.schedule') }}
        </span>
        <span class="contacts-list__schedule">{{ contacts.schedule }}</span>
      </span>
    </li>
  </ul>
</template>

<script>
import phoneConversion from '@/mixins/phone-conversion'

export default {
  mixins: [phoneConversion],
  computed: {
    contacts() {
      return this.$store.state.contacts
    }
  }
}
</script>

<style lang="scss">
.contacts-list {
  @include list-unstyled();

  &__item {
    display: flex;

    &:not(:last-of-type) {
      margin-bottom: 19px;
    }
  }

  &__icon {
    flex: 0 0 16px;
    width: 16px;
    height: 16px;
    margin-top: 3px;
    color: $primary;
  }

  &__description {
    display: flex;
    flex-direction: column;
    padding-left: 18px;
  }

  &__title {
    margin-bottom: 4px;
    font-size: 11px;
    line-height: 14px;
    color: $gray-dark-1;
  }

  &__link,
  &__schedule {
    font-size: 13px;
    font-weight: 500;
    line-height: 16px;
    letter-spacing: 0.05em;
  }

  @include media-breakpoint-up(md) {
    display: flex;
    flex-wrap: wrap;

    &__item {
      @include make-col(4);
      @include make-col-ready();

      flex-direction: column;
    }

    &__description {
      padding-left: 0;
    }

    &__icon {
      margin-top: 0;
      margin-bottom: 14px;
    }

    &__title {
      margin-bottom: 7px;
    }
  }
}
</style>
