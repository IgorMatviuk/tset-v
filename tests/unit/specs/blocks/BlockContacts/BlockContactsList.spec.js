import { mount, createLocalVue } from '@vue/test-utils'

import Vuex from 'vuex'

import BlockContactsList from '@/components/blocks/BlockContacts/BlockContactsList'

const localVue = createLocalVue()

localVue.use(Vuex)

describe('BlockContactsList', () => {
  let contacts
  let store

  beforeEach(() => {
    contacts = {
      phone: '0 800 332 512',
      email: 'office@virna.com.ua',
      schedule: 'с 9:00 - 18:00'
    }

    store = new Vuex.Store({
      state: {
        contacts
      }
    })
  })

  test('has expected structure', () => {
    const wrapper = mount(BlockContactsList, {
      localVue,
      store,
      stubs: ['svg-icon', 'client-only'],
      mocks: {
        $t: () => ''
      }
    })

    expect(wrapper.element).toMatchSnapshot()
  })

  test('render contacts list', () => {
    const wrapper = mount(BlockContactsList, {
      localVue,
      store,
      stubs: ['svg-icon', 'client-only'],
      mocks: {
        $t: () => ''
      }
    })

    expect(wrapper.findAll('.contacts-list__item').length).toEqual(
      Object.keys(store.state.contacts).length
    )
  })
})
