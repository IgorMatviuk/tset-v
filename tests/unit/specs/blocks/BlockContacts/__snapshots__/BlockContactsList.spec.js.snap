// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlockContactsList has expected structure 1`] = `
<ul
  class="contacts-list"
>
  <li
    class="contacts-list__item"
  >
    <svg-icon-stub
      class="contacts-list__icon"
      height="16"
      name="phone"
      width="16"
    />
     
    <span
      class="contacts-list__description"
    >
      <span
        class="contacts-list__title"
      >
        
        
      
      </span>
       
      <a
        class="contacts-list__link"
        href="tel:0800332512"
      >
        
        0 800 332 512
      
      </a>
    </span>
  </li>
   
  <li
    class="contacts-list__item"
  >
    <svg-icon-stub
      class="contacts-list__icon"
      height="16"
      name="mail"
      width="16"
    />
     
    <span
      class="contacts-list__description"
    >
      <span
        class="contacts-list__title"
      >
        
        
      
      </span>
       
      <a
        class="contacts-list__link"
        href="mailto:office@virna.com.ua"
      >
        
        office@virna.com.ua
      
      </a>
    </span>
  </li>
   
  <li
    class="contacts-list__item"
  >
    <svg-icon-stub
      class="contacts-list__icon"
      height="16"
      name="headset"
      width="16"
    />
     
    <span
      class="contacts-list__description"
    >
      <span
        class="contacts-list__title"
      >
        
        
      
      </span>
       
      <span
        class="contacts-list__schedule"
      >
        —Å 9:00 - 18:00
      </span>
    </span>
  </li>
</ul>
`;
